workflow/artifacts/s3/s3.go

```diff
@@ -38,13 +41,23 @@ func (s3Driver *S3ArtifactDriver) newS3Client() (argos3.S3Client, error) {
 func (s3Driver *S3ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
 	err := wait.ExponentialBackoff(wait.Backoff{Duration: time.Second * 2, Factor: 2.0, Steps: 5, Jitter: 0.1},
 		func() (bool, error) {
+			var err error
+			var bucket []byte
+			if inputArtifact.S3.BucketSecret.Key != "" {
+				bucket, err = ioutil.ReadFile(filepath.Join(common.SecretVolMountPath, inputArtifact.S3.BucketSecret.Name, inputArtifact.S3.BucketSecret.Key))
+				if err != nil {
+					return false, err
+				}
+			} else {
+				bucket = []byte(inputArtifact.S3.Bucket)
+			}
 			log.Infof("S3 Load path: %s, key: %s", path, inputArtifact.S3.Key)
 			s3cli, err := s3Driver.newS3Client()
 			if err != nil {
 				log.Warnf("Failed to create new S3 client: %v", err)
 				return false, nil
 			}
-			origErr := s3cli.GetFile(inputArtifact.S3.Bucket, inputArtifact.S3.Key, path)
+			origErr := s3cli.GetFile(string(bucket), inputArtifact.S3.Key, path)
```

Comment:
*Question* Any reason not to convert `bucket` to a `string` above on lines 47 or 52?